### PR TITLE
Enh output node

### PR DESCRIPTION
**Pull request by omar-rifai**

### CODE REVIEW

File: clinica/pipelines/t1_linear/t1_linear_utils.py [hunk 1]
```diff
@@ -1,3 +1,32 @@
+from clinica.utils.exceptions import ClinicaParserError
+
+
+def extract_source_entities(ref_file):
```
Comment (NicolasGensollen) [inline clinica/pipelines/t1_linear/t1_linear_utils.py:4]: I feel like these functions are not specific to the `t1_linear` pipeline, they extract entities from bids file name and vice-versa. They could probably be shared across pipelines. WDYT?

File: clinica/pipelines/t1_linear/t1_linear_utils.py [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+    import pathlib
+
+    # TODO: smarter entities extraction
+    source_entities = ("_").join(pathlib.Path(ref_file).stem.split("_")[:-1])
+    ext = "".join(pathlib.Path(ref_file).suffixes)
+
+    return source_entities, ext
+
+
+def construct_derivative_entities(pipeline_entities, pipeline_parameters):
```
Comment (NicolasGensollen) [inline clinica/pipelines/t1_linear/t1_linear_utils.py:14]: Could also be worth adding type hints for consistency (even though the other functions of this file do not seem to have them...).

File: clinica/pipelines/t1_linear/t1_linear_utils.py (hunk 1 above)
Comment (omar-rifai) [reply clinica/pipelines/t1_linear/t1_linear_utils.py:4]: Indeed, that is the goal ultimately. This currently is sort of a POC to see if it works :)

File: clinica/pipelines/t1_linear/t1_linear_utils.py (hunk 2 above)
Comment (omar-rifai) [reply clinica/pipelines/t1_linear/t1_linear_utils.py:14]: From what I remember, type hints did not couple well with Nipype Nodes - I can check again.

**Comment by omar-rifai** [comment]
Effort stalled for investigating Pydra route.

**Pull request closed without merging**